cmd/atlas/atlas.go

```diff
@@ -40,8 +44,14 @@ func Execute(ctx context.Context) {
 
 // initConfig reads in config file and ENV variables if set.
 func initConfig() {
-	if err := config.Load(); err != nil {
-		log.Fatalf("Error loading config: %v", err)
+	if err := config.LoadAtlasCLIConfig(); err != nil {
+		// we use mongocli.toml to generate atlasCLI config
+		var e viper.ConfigFileNotFoundError
+		if !errors.As(err, &e) || !createConfigFromMongoCLIConfig() { // search mongoCLI config only if atlasCLI config doesn't exist
+			printError(err)
```

Comment:
double casting of err into `viper.ConfigFileNotFoundError`

Reply:
it is wanted: In the `if` I check if the error is not `ConfigFileNotFoundError`: in this case, I have to print the error. If the error is `ConfigFileNotFoundError` (config not found), I need to run the `createConfigFromMongoCLIConfig`. This function may not be successful, in that case I have to check again if the error was  `ConfigFileNotFoundError` because we don't print error in that case

Reply:
this check was first inside `createConfigFromMongoCLIConfig` but it was moved here to address this https://github.com/mongodb/mongocli/pull/977#discussion_r804593264